src/datafile.rkt

```diff
@@ -29,27 +29,27 @@
                note
                tests))
 
-(define (try-list-accessor acc fail)
-  (λ (l) (if (null? l) fail (acc l))))
-
 (define (trs->pareto trs)
-  (define cas (map table-row-cost-accuracy trs))
-  (define starts (map (try-list-accessor first (list 0 0)) cas))
-  (define ptss (map (try-list-accessor (λ (ca) (cons (second ca) (third ca)))
-                                       (list (list 0 0)))
-                    cas))
-  (define reprs (map (compose get-representation table-row-precision) trs))
-
-  (define start
-    (for/fold ([x 0] [y 0] #:result (cons x y)) ([s starts])
-      (values (+ x (first s)) (+ y (second s)))))
-  (define ptss*
-    (for/list ([pts ptss])
-      (for/list ([pt pts])
-        (cons (first pt) (second pt)))))
-  (define ymax (apply + (map representation-total-bits reprs)))
-  (define frontier (map (λ (pt) (cons (first pt) (second pt))) (pareto-combine ptss #:convex? #t)))
-  (values start frontier ymax))
+  (match-let*
+      ([(cons start-cost start-accuracy)
+        (foldl (match-lambda*
+                 [(list r (cons x y))
+                  (match-let ([(list c a) ((compose first table-row-cost-accuracy) r)])
+                    (cons (+ x c) (+ y a)))])
+               (cons 0.0 0)
+               trs)]
+       [representations (map (compose get-representation table-row-precision) trs)]
+       [accuracy-max (foldl (lambda (x a) (+ a (representation-total-bits x))) 0 representations)]
+       [pareto-rescale (curry map (match-lambda [(list c a _ ...) (list (/ c start-cost) a)]))]
+       [rescaled
+        (map (compose
+              pareto-rescale
+              (lambda (ca) (cons (second ca) (third ca)))
```

Comment:
This makes no sense. Why do you want to have just the second and third cost-accuracy values? You just want to drop the first cost-accuracy point, since it is the input, right? Just use the `rest` function (https://docs.racket-lang.org/reference/pairs.html#%28def._%28%28lib._racket%2Flist..rkt%29._rest%29%29)